src/main/java/com/google/sps/data/UrlRequest.java

```diff
@@ -0,0 +1,78 @@
+package com.google.sps.data;
+
+import java.io.IOException;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.net.URLEncoder;
+import java.net.URL;
+import java.net.HttpURLConnection;
+import java.util.Map;
+import java.util.List;
+import java.util.ArrayList;
+
+/* Class conatining helper methods to make url requests */
+public final class UrlRequest {
+  /**
+   * Returns the a JSON string with the API response given an URL path and the query parameters.
+   */
+  public static String urlQuery(String basePath, Map<String, String> params) throws IOException {
```

Comment:
[Nit] This no longer throws an exception right?  In that case you can remove the throws clause at the end of the function.

Reply:
The url handling and connection can still throw an IOException.